ckanext/unaids/assets/js/validation-badge.js

```diff
@@ -0,0 +1,55 @@
+this.ckan.module('validation-badge', function (jQuery) {
+    return{
+        options: {
+          resource: null,
+          status: 'created',
+          url: null
+        },
+        initialize: function () {
+          $.proxyAll(this, /_on/);
+          this._poll();
+          this.options.url = $(this.el).attr('href');
+          $(this.el).removeAttr('href');
+        },
+        _poll: function() {
+            var module = this;
+            $.ajax({
+                url: "/api/3/action/resource_validation_show?resource_id=" + this.options.resource,
+                type: "GET",
+                success: function(data){module._success(data);},
+                error: function(XMLHttpRequest, textStatus, errorThrown){module._error(XMLHttpRequest, textStatus, errorThrown);},
+                dataType: "json",
+                complete: setTimeout(function(){module._complete();}, 5000),
+                timeout: 2000
+            });
+        },
+        _success: function(data) {
+            this.options.status = data.result.status;
+            if(this.options.status != 'running' && this.options.status != 'created'){
```

Comment:
This doesn't seem to work at all for me - if I have a validation that is running (e.g. a large geojson) then I get no badge (which is the upstream behaviour)

Reply:
Could you share with me a working geojson file so I can test it please?